job/server/src/main/java/alluxio/master/job/JobMasterClientServiceHandler.java

```diff
@@ -85,7 +86,12 @@ public void getJobServiceSummary(GetJobServiceSummaryPRequest request,
   @Override
   public void listAll(ListAllPRequest request, StreamObserver<ListAllPResponse> responseObserver) {
     RpcUtils.call(LOG, (RpcUtils.RpcCallableThrowsIOException<ListAllPResponse>) () -> {
-      return ListAllPResponse.newBuilder().addAllJobIds(mJobMaster.list()).build();
+      ListAllPResponse.Builder builder = ListAllPResponse.newBuilder()
+          .addAllJobIds(mJobMaster.list());
+      for (JobInfo jobInfo : mJobMaster.listDetailed()) {
+        builder.addJobInfos(jobInfo.toProto());
+      }
```

Comment:
Is getting/returning the detailed info expensive? If so we could make it an option in the `listAll` API and only attach it if specifically requested.

Reply:
From what I looked, it's not very expensive. The only reason I kept around the listAll() old version is just for backcompat.